src/iop/watermark.c

```diff
@@ -718,6 +720,38 @@ static gchar *_watermark_get_svgdoc(dt_iop_module_t *self, dt_iop_watermark_data
       g_list_free_full(res, &g_free);
     }
 
+    res = dt_tag_get_list(image->id);
+    gchar *keywords = dt_util_glist_to_str(", ", res);
+    svgdoc = _string_substitute(svgdata, "$(IMAGE.TAGS)", (keywords ? keywords : ""));
+    if(svgdoc != svgdata)
+    {
+      g_free(svgdata);
+      svgdata = svgdoc;
+    }
+    if(keywords)
+    {
+      g_free(keywords);
+    }
+    if(res)
+    {
+      g_list_free_full(res, &g_free);
+    }
+
+    res = dt_metadata_get(image->id, "Xmp.xmp.Rating", NULL);
```

Comment:
I would use `image->flags` instead, that should be a little faster. Have a look at `metadata_view.c` to see how the rating is stored in there.

Reply:
Revised code to employ `image->flags`.